smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/json/MemberShapeDecodeGenerator.kt

```diff
@@ -154,7 +155,7 @@ abstract class MemberShapeDecodeGenerator(
      */
     open fun renderAssigningDecodedMember(topLevelMember: MemberShape, decodedMemberName: String) {
         val topLevelMemberName = ctx.symbolProvider.toMemberName(topLevelMember)
-        writer.write("\$L = \$L", topLevelMemberName, decodedMemberName)
+        writer.write("\$L\$L = \$L", path, topLevelMemberName, decodedMemberName)
```

Comment:
Related to other comment I left - should there be some condition that checks that error is embedded in event streams?

Reply:
Not needed here, this code is not called and this decoder is not generated unless the error is a member of an event stream.